node_modules/col-assets/lib/api.js

```diff
@@ -513,7 +517,25 @@ var getAssets = module.exports.getAssets = function(ctx, filters, sort, limit, o
           });
         });
 
-        return callback(null, data);
+        var options = {
+          'where': {
+            'asset_id': assetIds
+          }
+        };
+        DB.Pin.findAll(options).complete(function(err, pins) {
```

Comment:
@johncrossman Thanks for trying things out; and agreed, let's create a new Jira for any further work in that direction.

Per comments in code, I think the nested `DB.User.findAll` is necessary because it retrieves all associated users even when we're filtering by a given user: so it may return more information than was present in the original query.